core/src/key.rs

```diff
@@ -0,0 +1,99 @@
+use std::{fs, path::PathBuf, str::FromStr};
+
+use anyhow::{anyhow, Result};
+use cid::multihash::Multihash;
+use cid::Cid;
+use ring::signature::{Ed25519KeyPair, KeyPair};
```

Comment:
I'm wondering which library we should expose here.. we use [ed25519_dalek](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/) in some places as ssi uses it. It has `SigningKey::from_bytes` (for both secret and public key options) but I'm not sure it's really that much nicer.. mostly just curious what we _should_ be using as our public API.